psi4/src/psi4/libmints/mintshelper.cc

```diff
@@ -1995,6 +1996,171 @@ SharedMatrix MintsHelper::perturb_grad(SharedMatrix D, double xlambda, double yl
     return ret;
 }
 
+SharedMatrix MintsHelper::dipole_grad(SharedMatrix D) {
+    // Computes skeleton (Hellman-Feynman like) dipole derivatives for each perturbation
+    double **Dp = D->pointer();
+
+    int natom = molecule_->natom();
+    auto ret = std::make_shared<Matrix>("Dipole dervatives (pert*component, i.e. 3Nx3)", 3 * natom, 3);
+    double **Pp = ret->pointer();
+
+    std::shared_ptr<OneBodyAOInt> Dint(integral_->ao_dipole(1));
+    const double *buffer = Dint->buffer();
+
+    for (int P = 0; P < basisset_->nshell(); P++) {
```

Comment:
I think we copy/paste this code a bit. Is it possible to make a general function that takes a OneBodyAOInt derivative's to produce this quantity?